File: src/main/java/com/gitlab/daring/image/command/env/ShowCommand.java
package com.gitlab.daring.image.command.env;

import com.gitlab.daring.image.command.BaseCommand;
import com.gitlab.daring.image.command.CommandEnv;
import com.gitlab.daring.image.component.BaseCanvasFrame;

import static com.gitlab.daring.image.command.CommandScriptUtils.runCommand;
import static javax.swing.WindowConstants.DISPOSE_ON_CLOSE;

public class ShowCommand extends BaseCommand {

    final String title = arg(0, "");
    final String key = arg(1, "");
    final BaseCanvasFrame frame = new BaseCanvasFrame(title);

    public ShowCommand(String... params) {
        super(params);
        if (!isCacheable()) frame.setDefaultCloseOperation(DISPOSE_ON_CLOSE);
    }

    @Override
    public boolean isCacheable() {
        return !title.isEmpty();
    }

    @Override
    public void execute(CommandEnv env) {
        String tv = env.eval(title);
        if (title.equals(tv)) {
            frame.showMat(env.getMat(key));
        } else {
            runCommand(env, "show", tv, key);
        }
    }

    @Override
    public void close() {
        frame.dispose();
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/env/ShowCommand.kt
package com.gitlab.daring.image.command.env

import com.gitlab.daring.image.command.CommandEnv
import com.gitlab.daring.image.command.CommandScriptUtils.runCommand
import com.gitlab.daring.image.command.KBaseCommand
import com.gitlab.daring.image.swing.BaseCanvasFrame
import javax.swing.WindowConstants.DISPOSE_ON_CLOSE

internal class ShowCommand(args: Array<String>) : KBaseCommand(args) {

    val title = arg(0, "")
    val key = arg(1, "")
    val frame = BaseCanvasFrame(title)

    init {
        if (!isCacheable) frame.defaultCloseOperation = DISPOSE_ON_CLOSE
    }

    override val isCacheable get() = !title.isEmpty()

    override fun execute(env: CommandEnv) {
        val tv = env.eval(title)
        if (title == tv) {
            frame.showMat(env.getMat(key))
        } else {
            runCommand(env, "show", tv, key)
        }
    }

    override fun close() {
        frame.dispose()
    }

}